app/controllers/users/piv_cac_recommended_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Users
+  class PivCacRecommendedController < ApplicationController
+    include TwoFactorAuthenticatableMethods
+    include MfaSetupConcern
+    include SecureHeadersConcern
+    include ReauthenticationRequiredConcern
+
+    before_action :authenticate_user!
+    before_action :confirm_user_authenticated_for_2fa_setup
```

Comment:
`authenticate_user!` happens inside `confirm_user_authenticated_for_2fa_setup`, so do we need it twice?

https://github.com/18F/identity-idp/blob/83ab9b147f6311fe476fb2fefd8aaac6eae075a6/app/controllers/concerns/mfa_setup_concern.rb#L36-L37

```suggestion
    before_action :confirm_user_authenticated_for_2fa_setup
```

Reply:
Although this raises another concern, which is that we'd be prompting someone who's signing in with remembered device to reauthenticate. Shouldn't we only reauthenticate them when they choose to add PIV as an authentication method, and not before then?